pkg/controller/dnszone/dnszone_controller_test.go

```diff
@@ -141,20 +141,6 @@ func TestReconcileDNSProviderForAWS(t *testing.T) {
 				assert.NotNil(t, condition, "zone available condition should be set on dnszone")
 			},
 		},
-		{
-			name:    "Cannot create a hosted zone as credentials are missing permissions",
-			dnsZone: validDNSZoneWithoutID(),
-			setupAWSMock: func(expect *mock.MockClientMockRecorder) {
-				mockAccessDeniedException(expect, validDNSZoneWithoutID())
-			},
-			validateZone: func(t *testing.T, zone *hivev1.DNSZone) {
-				condition := controllerutils.FindDNSZoneCondition(zone.Status.Conditions, hivev1.InsufficientCredentialsCondition)
-				assert.NotNil(t, condition, "invalid credentials condition should be set on dnszone")
-				assert.Equalf(t, "User: arn:aws:iam::0123456789:user/testAdmin is not authorized to perform: tag:GetResources with an explicit deny",
-					condition.Message, "condition has an unexpected message")
-			},
-			errorExpected: true,
-		},
```

Comment:
For unit tests I might have to mock `getActuator` method of dnszone controller and to do that  I will have to make it a function variable in `ReconcileDNSZone` struct (similar to [soaLookup](https://github.com/openshift/hive/blob/3261cec3ba098e1ca4019db3d1b91a2db86f95c9/pkg/controller/dnszone/dnszone_controller.go#L111) method).

Reply:
That is a fine way if you need it. Will it still be necessary once https://github.com/openshift/hive/pull/1055 lands, which moves the code for setting conditions based on errors into the individual actuators?

Reply:
Actually, I wonder if this entire PR may not be necessary given the changes in #1055.

Reply:
Other actuator methods can also throw access denied errors, so I thought of moving this check outside `reconcileDNSProvider`. May be I can use `SetConditionsForError` for each actuator method as done in #1055.

Reply:
Ah, yes. I got confused at though that #1055 was doing that, too.